lua/luasnip/util/environ.lua

```diff
@@ -21,6 +21,33 @@ local function _fill_eager_vars(env, pos)
 end
 
 local lazy_vars = {}
+local env_namespaces = {}
+
+-- Namespaces allow users to define their own environmet variables
+local function _resolve_namespace_var(full_varname)
+	local parts = vim.split(full_varname, "_")
+	if #parts < 2 then
+		return nil
+	end
+	local nmsp = env_namespaces[parts[1]]
+
+	if not nmsp then
+		return nil
+	end
+
+	local varname = full_varname:sub(#parts[1] + 2)
+
+	if type(nmsp) == "function" then
```

Comment:
Nice, this is better than requiring the user to create a table even if they only need (basically) `__index` :+1:
I think this could be even nicer by making the namespaces exclusively functions (if a table is passed it is wrapped inside a function).